### PR TITLE
Add initial TextureArray wrapper

### PR DESCRIPTION
**Pull request by Tom-Ski**
Adds support for TextureArray objects with LibGDX.

This requires GLES 3.0+ compatibility, and has been tested on android/lwjgl and lwjgl3 backends.
A few tweaks were needed to the android and lwjgl graphics interfaces to correctly create an uninitialized 3D texture.

See the sample test case provided for usecase.

### CODE REVIEW

File: gdx/src/com/badlogic/gdx/graphics/TextureArrayData.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+
```
Comment (intrigus) [inline gdx/src/com/badlogic/gdx/graphics/TextureArrayData.java]: Missing license header

File: gdx/src/com/badlogic/gdx/graphics/glutils/FileTextureArrayData.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	@Override
+	public boolean isPrepared () {
+		return prepared;
+	}
+
+	@Override
+	public void prepare () {
+		int width = -1;
+		int height = -1;
+		for (TextureData data : textureDatas) {
```
Comment (intrigus) [inline gdx/src/com/badlogic/gdx/graphics/glutils/FileTextureArrayData.java]: Nitpicking: I'd either use an libgdx Array or iterate over textureDatas using a `for(int i....)` loop, so we don't create (unnecessary) garbage.

File: gdx/src/com/badlogic/gdx/graphics/glutils/FileTextureArrayData.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	@Override
+	public int getGLType () {
+		return Pixmap.Format.toGlType(format);
+	}
+
+	@Override
+	public boolean isManaged () {
+		for (TextureData data : textureDatas) {
```
Comment (intrigus) [inline gdx/src/com/badlogic/gdx/graphics/glutils/FileTextureArrayData.java]: Same here

File: gdx/src/com/badlogic/gdx/graphics/TextureArray.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+		bind();
+		Gdx.gl30.glTexImage3D(GL30.GL_TEXTURE_2D_ARRAY, 0, data.getInternalFormat(), data.getWidth(), data.getHeight(), data.getDepth(), 0, data.getInternalFormat(), data.getGLType(), null);
+
+		if (!data.isPrepared()) data.prepare();
+
+		data.consumeTextureArrayData();
+
+		setFilter(minFilter, magFilter);
+		setWrap(uWrap, vWrap);
```
Comment (xoppa) [inline gdx/src/com/badlogic/gdx/graphics/TextureArray.java]: You can use `unsafeSetFilter`, `unsafeSetWrap` here

**Comment by xoppa** [comment]
Looks good! I'd say merge it (and update CHANGES)

**Comment by magicgoose** [comment]
Do you have plans to integrate it with TextureAtlas? (so that TextureAtlas would use 1 TextureArray instead of several textures)

**Pull request merged**